lib/dom.js

```diff
@@ -606,14 +606,27 @@ Document.prototype = {
 		return rtv;
 	},
 	
-	getElementsByClassName: function(className) {
-		var pattern = new RegExp("(^|\\s)" + className + "(\\s|$)");
+	getElementsByClassName: function(classNames) {
+		var spaceSeparatedTokens = function(string) {
+			// 5 space characters, as defined in html spec
+			return string.split(/[ \t\n\f\r]+/).filter(function(name) {
+				return name !== ''
+			})
```

Comment:
We are skipping a specific part of the spec in this implementation, that says that the values should not contain duplicates: https://dom.spec.whatwg.org/#concept-ordered-set-parser (it's "hidden" behind the "ordered set" link).

Since we can not rely on `Set` being implemented, the easiest way to avoid duplicates is to use the keys of a dict (object literal).

If you need more details or don't want to bother, I could work on this part after the other changes have been applied. But I would rather add it to this PR, since it basically "adds support for multiple class names".